sdks/java/io/rabbitmq/src/main/java/org/apache/beam/sdk/io/rabbitmq/RabbitMqMessage.java

```diff
@@ -0,0 +1,297 @@
+    return headers;
+  }
+
+  public Integer getDeliveryMode() {
+    return deliveryMode;
+  }
+
+  public Integer getPriority() {
+    return priority;
+  }
+
+  public String getCorrelationId() {
+    return correlationId;
+  }
+
+  public String getReplyTo() {
+    return replyTo;
+  }
+
+  public String getExpiration() {
+    return expiration;
+  }
+
+  public String getMessageId() {
+    return messageId;
+  }
+
+  public Date getTimestamp() {
+    return timestamp;
+  }
+
+  public String getType() {
+    return type;
+  }
+
+  public String getUserId() {
+    return userId;
+  }
+
+  public String getAppId() {
+    return appId;
+  }
+
+  public String getClusterId() {
+    return clusterId;
+  }
+
+  public AMQP.BasicProperties createProperties() {
+    return new AMQP.BasicProperties()
+        .builder()
+        .contentType(contentType)
+        .contentEncoding(contentEncoding)
+        .headers(headers)
+        .deliveryMode(deliveryMode)
+        .priority(priority)
+        .correlationId(correlationId)
+        .replyTo(replyTo)
+        .expiration(expiration)
+        .messageId(messageId)
+        .timestamp(timestamp)
+        .type(type)
+        .userId(userId)
+        .appId(appId)
+        .clusterId(clusterId)
+        .build();
+  }
+
+  @Override
+  public int hashCode() {
+    return Objects.hash(
+        routingKey,
+        Arrays.hashCode(body),
+        contentType,
+        contentEncoding,
+        deliveryMode,
+        priority,
+        correlationId,
+        replyTo,
+        expiration,
+        messageId,
+        timestamp,
+        type,
+        userId,
+        appId,
+        clusterId);
+  }
+
+  @Override
+  public boolean equals(Object obj) {
+    if (obj instanceof RabbitMqMessage) {
+      RabbitMqMessage other = (RabbitMqMessage) obj;
+      if (routingKey == null
+          && other.routingKey == null
+          && body == null
+          && other.body == null
+          && contentType == null
+          && other.contentType == null
+          && contentEncoding == null
+          && other.contentEncoding == null
+          && deliveryMode == null
+          && other.deliveryMode == null
+          && priority == null
+          && other.priority == null
+          && correlationId == null
+          && other.correlationId == null
+          && replyTo == null
+          && other.replyTo == null
+          && expiration == null
+          && other.expiration == null
+          && messageId == null
+          && other.messageId == null
+          && timestamp == null
+          && other.timestamp == null
+          && type == null
+          && other.type == null
+          && userId == null
+          && other.userId == null
+          && appId == null
+          && other.appId == null
+          && clusterId == null
+          && other.clusterId == null) {
+        return true;
+      } else {
+        boolean equals = false;
+        if (routingKey != null) {
```

Comment:
Use Objects.equals for all of these - check how equals() is implemented on other classes in Beam